lib/src/Uncertainty/Distribution/TruncatedDistribution.cxx

```diff
@@ -173,25 +174,37 @@ TruncatedDistribution * TruncatedDistribution::clone() const
 }
 
 /* Get the simplified version (or clone the distribution) */
-Distribution TruncatedDistribution::getSimplifiedVersion() const
+Bool TruncatedDistribution::hasSimplifiedVersion(Distribution & simplified) const
 {
+  // n-D case
+  JointDistribution *p_joint = dynamic_cast<JointDistribution *>(distribution_.getImplementation().get());
+  if (p_joint && p_joint->hasIndependentCopula() && (getDimension() > 1))
+  {
+    Collection<Distribution> coll(getDimension());
+    for (UnsignedInteger i = 0; i < getDimension(); ++ i)
+      coll[i] = TruncatedDistribution(p_joint->getMarginal(i), bounds_.getMarginal(i));
+    simplified = JointDistribution(coll);
+    return true;
+  }
+
   // Simplification of the 1D case
   Distribution localDistribution(distribution_);
   String kind(localDistribution.getImplementation()->getClassName());
   // Delve into the antecedents until we get something which is not a truncated distribution
+  UnsignedInteger level = 1;
   while (kind == "TruncatedDistribution")
   {
     const TruncatedDistribution * truncatedDistribution = dynamic_cast<const TruncatedDistribution *>(localDistribution.getImplementation().get());
     localDistribution = truncatedDistribution->getDistribution();
     kind = localDistribution.getImplementation()->getClassName();
+    ++ level;
```

Comment:
Maybe you could add a comment on the fact that the nested truncation intervals have already been intersected during the nested range computations.